sdk/websites/Azure.ResourceManager.AppService/tests/TestsCase/SiteCollectionTests.cs

```diff
@@ -23,6 +23,37 @@ private async Task<WebSiteCollection> GetSiteCollectionAsync()
             return resourceGroup.GetWebSites();
         }
 
+        [TestCase]
+        [RecordedTest]
+        public async Task Demo()
+        {
+            var container = await GetSiteCollectionAsync();
+            var name = Recording.GenerateAssetName("testSite");
+            var input = ResourceDataHelper.GetBasicSiteData(DefaultLocation);
+            var lro = await container.CreateOrUpdateAsync(WaitUntil.Completed, name, input);
+            var site = lro.Value;
+
+            int ftp = 0;
+            int scm = 0;
+            await foreach (var policy in site.GetBasicPublishingCredentialsPoliciesAsync())
+            {
+                // unfortunately all the element here is of type of BaseWebSitePublishingCredentialsPolicyResourceProxy
```

Comment:
most likely need to use a dynamic type here https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/Instrumentation/AsyncPageableInterceptor.cs#L22.

probably `_inner.Current.GetType()` instead of `typeof(T)`